ethexe/cli/src/service.rs

```diff
@@ -781,6 +800,29 @@ impl Service {
             }
         }
     }
+
+    async fn process_response_validation(
+        validating_response: &db_sync::ValidatingResponse,
+        router_query: &mut RouterQuery,
+    ) -> Result<bool> {
+        let response = validating_response.response();
+
+        if let db_sync::Response::ProgramIds(ids) = response {
+            let ethereum_programs = router_query.programs_count().await?;
+            if ethereum_programs != U256::from(ids.len()) {
+                return Ok(false);
+            }
+
+            for &id in ids {
+                let code_id = router_query.program_code_id(id).await?;
```

Comment:
wdyt about at least putting issue for multiple queries? like by batches of 32 ids mb?